src/SSHDebugPS/CommandRunner.cs

```diff
@@ -9,304 +9,304 @@
+            : this(settings.Command, settings.CommandArgs)
+        { }
 
-        public void Run()
+        public LocalCommandRunner(string command, string commandArgs)
         {
-            _localProcess = new System.Diagnostics.Process();
-            _localProcess.StartInfo = processStartInfo;
-            _localProcess.Exited += OnProcessExited;
-            _localProcess.EnableRaisingEvents = true;
-            _localProcess.Start();
-
-            _stdoutWriter = new StreamWriter(_localProcess.StandardInput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), 4096, leaveOpen: false);
-            _processReader = new StreamReader(_localProcess.StandardOutput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), false, BUFMAX, false);
-            _outputReadTask = System.Threading.Tasks.Task.Run(() => ReadLoop(_processReader, _cancellationSource.Token, (line) => { OutputReceived?.Invoke(this, line); }));
-
-            _localProcess.ErrorDataReceived += OnErrorOutput;
-            _localProcess.Exited += OnProcessExited;
-            _localProcess.BeginErrorReadLine();
+            CreateProcessStartInfo(command, commandArgs);
         }
 
-        private static int BUFMAX = 4096;
-        private void ReadLoop(StreamReader reader, CancellationToken token, Action<string> action)
+        public void Start()
         {
-            try
+            if (_process != null)
             {
-                while (!token.IsCancellationRequested)
+                if (!_process.HasExited)
                 {
-                    char[] buffer = new char[BUFMAX];
-                    Task<string> task = reader.ReadLineAsync();
-                    task.Wait(token);
+                    Debug.Fail("Process is already running.");
 
-                    if (task.Result == null)
-                    {
-                        lock (_lock)
-                        {
-                            if (!_hasExited && _localProcess.HasExited)
-                            {
-                                _hasExited = true;
-                                Closed?.Invoke(this, _localProcess.ExitCode);
-                            }
-                        }
-                        return; // end of stream
-                    }
-
-                    action(task.Result);
+                    throw new InvalidOperationException("Process already running");
                 }
+                CleanUpProcess();
             }
-            catch (Exception e)
+
+            if (_processStartInfo == null)
             {
-                ErrorOccured?.Invoke(this, new ErrorOccuredEventArgs(e));
-                Dispose();
+                throw new InvalidOperationException("Unable to create process. Process start info does not exist");
             }
-        }
 
-        private void OnErrorOutput(object sender, DataReceivedEventArgs e)
-        {
-            if (!string.IsNullOrWhiteSpace(e.Data))
+            lock (_lock)
```

Comment:
>lock (_lock) [](start = 12, length = 12)

I think this lock should be around the entire method - otherwise, if you could end up with two threads both sitting at this point, where the first one gets the lock, initializes _process, and releases the lock, at which point the second thread will overwrite _process.